pyproject.toml

```
@@ -102,6 +102,11 @@ dependencies = [
   "tqdm<5",
   'typing-extensions<5; python_version <= "3.7"',
   "uvicorn<0.25",
+  "yarl",
+  "numpy",
+  "pyyaml",
+  "pandas",
+  "pyarrow"
```

I would have expected the linter croaks because the dependencies have not been sorted correctly. Anyway. However, please let's add corresponding version limiters.
```suggestion
  "yarl<1.10",
  "numpy<1.27",
  "pandas<3",
  "pyarrow<17",
  "pyyaml<7",
```

---

Sure can do. Just a reminder, this is a draft not a final PR ;)